app/models/forms/v2/LiabilityDetailsForm.scala

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.forms.v2
+
+import java.time.Instant
+
+import models.forms.FormConstraints.{dateMustBeInThePast, emptyOr}
+import models.forms.FormDate
+import models.forms.mappings.Constraints
+import models.forms.mappings.FormMappings._
+import models._
+import play.api.data.Form
+import play.api.data.Forms._
+
+object LiabilityDetailsForm extends Constraints {
+
+  private val numbersOnlyRegex = """^\d+$""".r
+  private val btiRefRegex = """[0-9]{6,22}""".r
```

Comment:
Maybe we should put validation regexes for things relating to ATaR (had to look that up) data in a utility class?

Reply:
e.g. similar to `FormDate`

Reply:
extract these lines into methods for each case in form constraints